cmd/action/ci/ci.go

```diff
@@ -133,7 +134,13 @@ func setCfgName(cfg *v1alpha1.Environment) {
 	// uid is unique for each run
 	uid := generateUID()
 
-	cfg.Name = fmt.Sprintf("ci%s-%s-%s", attempt, sha, uid)
+	actor := os.Getenv("GITHUB_ACTOR")
+	branchName := os.Getenv("GITHUB_REF_NAME")
+	repo_name := os.Getenv("GITHUB_REPOSITORY")
+	parts := strings.Split(repo_name, "/")
+	repoName := parts[len(parts)-1]
+
+	cfg.Name = fmt.Sprintf("ci%s-%s-%s-%s-%s-%s", attempt, actor, branchName, repoName, sha, uid)
```

Comment:
resultant gpu-driver-container e2e instance name with this change : 

Actor: shivakunv
gpu-driver-container Branch Name:  holodeckname
repo: gpu-driver-container 

ci5-shivakunv-holodeckname-gpu-driver-container-a0917fc6-fqpz5zu8

Reply:
We should add these metadata as tags to the AWS resources namely - the VPC, subnets and the EC2 instance

[AWS Tags docs](https://docs.aws.amazon.com/whitepapers/latest/tagging-best-practices/what-are-tags.html).

Reply:
Agree with Tariq, let's add these attributes as tags for each resource that holodeck creates.

I am fine to update the ec2 resource name, but right now it is a bit verbose. What about just `ci<attempt>-<repoName>-<sha>`?

Reply:
@tariq1890 cf.name is already configured for vpcs .  working on other tags

@cdesiniotis   This code was already  there :- 
cfg.Name = fmt.Sprintf("ci%s-%s-%s", attempt, sha, uid)    

but we were not able to identify the owner ,repo branch with information by looking , we have to search commit_sha in this case .

Reply:
@shivakunv For this PR, let the `cfg.Name` remain as-is.  The Name should never hold the metadata. it just needs to be unique, so the current Name components fulfill that requirement. 

The following should just be added as AWS tags [here](https://github.com/NVIDIA/holodeck/blob/main/pkg/provider/aws/aws.go#L98)

```
	actor := os.Getenv("GITHUB_ACTOR")
	branchName := os.Getenv("GITHUB_REF_NAME")
	repo_name := os.Getenv("GITHUB_REPOSITORY")
```

We can also add the following as additional metadata
```
GITHUB_RUN_ID
GITHUB_RUN_NUMBER
GITHUB_JOB
```

Reply:
added